java/core/src/test/org/apache/orc/impl/TestRecordReaderImpl.java

```diff
@@ -2449,4 +2449,23 @@ public void testSargApplier() throws Exception {
     f2.setAccessible(true);
     assertFalse((boolean)f2.get(applier1));
   }
+
+  @Test
+  public void testRgEndOffset() throws IOException {
+    Configuration conf = new Configuration();
+    Path filePath = new Path(getClass().getClassLoader().
+        getSystemResource("orc-small-compression-size.orc").getPath());
+    Reader reader = OrcFile.createReader(filePath, OrcFile.readerOptions(conf));
+
+    SearchArgument sarg = SearchArgumentFactory.newBuilder()
+        .startNot()
+        .lessThan("x", PredicateLeaf.Type.LONG, 120000L)
+        .end().build();
+    VectorizedRowBatch batch = reader.getSchema().createRowBatch();
+    try (RecordReader rows = reader.rows(reader.options().searchArgument(sarg, null))) {
+      while (rows.nextBatch(batch)) {
+        // do nothing but check all the data can be read
```

Comment:
Could you check the actual values to ensure they are read correctly?

Reply:
+1 -- it would be much easier if we generate the data as well making sure there are no inconsistencies